### DISCUSSION TITLE
Guidance on creating overlapping lower-resolution S102 "overview" products to aid PPU/ECDIS performance when zooming out?

### DISCUSSION THREAD

**Discussion by anthonyklemm** [iho-ohi/S-102-Product-Specification#85] [General]
As described in S102PT16, NOAA's early S102 testers (pilots using PPUs in five port areas of the US) have described PPU performance issues when zooming out to view the greater port area. We believe it is due to the software dealing with the high-computational load of drawing the high-resolution S102 bathymetry over a large area at its native resolution. 

As a workaround, NOAA has started to produce shoal-biased lower resolution S102 products to be viewable when the user zooms out to improve the user experience, similar to when a smaller scale ENC appears when there is a more appropriate scale band available. 

NOAA would like guidance on how these S102 “overviews” should be managed. Here a few questions:

- Should the lower resolution grids be embedded in the S102 file (similar to raster pyramiding), or should there be separate S102 files with different resolutions?
- How does the ECDIS or PPU determine when to switch from one resolution to another (similar to switching scale bands of an ENC when the user zooms in or out)?
- Is this something the S102PT should dictate in the product specification, or does this happen with S-98? (It seems ideal that they would transition in concert with the S101 scale bands)
- Should the PPU or ECDIS create the lower resolution “pyramids” upon loading the cells, instead of the HO creating them?
- Should the S102 cells just be removed from the display entirely if zoomed out too far (removing the need for lower resolution S102 overviews)? What is the cut-off view scale/zoom level? Is it resolution dependent?

![S-102 lower resolution overview products](https://github.com/iho-ohi/S-102-Product-Specification/assets/76973843/e52093c6-8917-4273-b011-d8ff34f15950)

**Comment by RohdeBSH** [comment]
Hi Anthony,

Very interesting questions. I don't have an answer either. What I don't like is saving the lower resolution grids in the same S-102 file. If I reduce the resolution, I may want to increase the product extent.
What comes to my mind spontaneously is the ExchangeSet. At least one "navigationPurpose" (port, transit, overview) is specified there for each S-102 file. This could be used by the ECDIS/PPU to prevent the S-102 from being displayed in the wrong scaleband.


This can perhaps be investigated in the BalticSea eNavigation project. I will point this out to those involved.

**Reply by anthonyklemm** [reply]
I've thought about this some more and think that maybe it is best that overviews are constructed on the client side. For example, I can now easily download several S102 tiles and use the gdal bindings in python to create a vrt and designate the overview resolutions it creates. I think that may be a good solution to this problem.

**Reply by RohdeBSH** [reply]
I think that with this issue we are also reaching the limits of the responsibility of the data producers (HOs). It is a practical portrayal issue that the OEMs should deal with. Because I don't know what hardware or software the OEMs are using.

**Reply by RohdeBSH** [reply]
You can also set "optimumDisplayScale", "maximumDisplayScale" & "minimumDisplayScale" in the ExcahngeSet "dataCoverage"-Attribute (Class S100_DataCoverage).
But I don't know how the OEMs handle the ExchangeSets. Is the information only read once or is it always available? That's why I prefer to have such data in the HDF5 file instead of outside.

**Comment by giumas** [comment]
My feeling is that each S-102 should be 'linked' to a corresponding S-101 by sharing the same display scale parameters.
Any other solution is surely doable, but likely to create troubles when interoperating with other S-100 products (e.g., S-104).

**Reply by RohdeBSH** [reply]
As far as I can see, neither the HDF5 file nor the ExchangeSet provides such a link.
Adding the link to S-100 5.2.0 is too late. In S-102 ed 3.0.0, it would be possible to include it in the ExchangeSet via a derived class of the "S100_DatasetDiscoveryMetadata" class. However, this is again a huge effort. It also makes the S-102 extremely unique and difficult to maintain.

The questions I also have are:

1. If I have multiple S-101 ENCs in different scale bands in the same place, do I also need to create multiple S-102s in different resolutions?
2. Do I have to take the possible depth areas of the S-101 into account when creating the S-102? 
We normally create the S-102 products first and then update the ENCs. This doesn't really fit together in the creation process.

**Reply by tfilppula** [reply]
Our S-102 products will sometimes likely span more than one S-101 ENCs - not covering the whole area of even one S-101 ENC but cross over two or more S-101 ENC coverages. In the attached example there is an area covered by a small test product (in red, S-102 file size ~1MB) that falls inside two ENC cells. In cases like this there is not one "corresponding" ENC cell, but we always have one or more ENCs with corresponding display scales. Maybe "linking", if necessary, should be based on display scales instead of ENC cell boundaries?

![image](https://github.com/iho-ohi/S-102-Product-Specification/assets/13709365/59304841-45d2-4102-8868-fde993b9b668)

**Reply by giumas** [reply]
I believe that the 'linking' can be to 1 or many ENCs as well as to 1 or more navigation purposes. 

In other words, without any display scale parameters, the S-102 is implicitly linked to all the overlapping ENCs.
If the display scale parameters are used, then the S-102 is linked to all the ENCSs (and related navigation purposes) in the selected display scale range.

It may be not necessary to change S-100 5.2.0 because the display scale parameters are already present (optional).

**Reply by RohdeBSH** [reply]
I think using "optimumDisplayScale", "maximumDisplayScale" & "minimumDisplayScale" in the ExcahngeSet "dataCoverage"-Attribute (Class S100_DataCoverage) is the easiest way to handle the issue. If you're doing so, you don't have to link one or more ENCs to transport the same information. This also doesn't require a modification of S-100 or S-102. You can just use it right away.
